### ISSUE TITLE
Support command links in notifications

### ISSUE THREAD

**Issue by eamodio** [Microsoft/vscode#46571]
Now that notifications support links via markdown (which should be documented), it would be great if there was support for command links.

Can probably reuse `MarkdownString` if these will need to be trusted in the same way as hovers.

//cc @jrieken @bpasero

**Comment by bpasero** [comment]
I am not sure about `MarkdownString` because that might draw the wrong assumption that notifications support any markdown. Currently we only support links (http / https) and I would stay away from opening that too much to ensure notifications always have a consistent look. 

I agree though that currently the fact that links are supported is a bit magic and hidden.

**Comment by eamodio** [comment]
Yeah, definitely agree on the look/styling (although bold and italic might be nice), but command links could be valuable. For example, I am adding a notice to GitLens if you have the old set of keyboard shortcuts (which are known to conflict with other keyboard layouts), and I attempted to provide a link to open the *GitLens Settings* editor if someone wanted to jump there.

Something like this:
![image](https://user-images.githubusercontent.com/641685/37921159-fbccc8e4-30f6-11e8-9e22-1c6e14792303.png)

**Comment by eamodio** [comment]
It would be great if not only `command:` links were supported but also registered uri handlers. I attempted to register a Uri handler to provide a link from a notification, but it looks like only `http|https` is supported.

**Comment by eamodio** [comment]
So if I wanted to submit a PR for this -- I assume just adding `vscode` and `vscode-insiders` to this regex would be fine. https://github.com/Microsoft/vscode/blob/b2a361e9899cd20679a7592e1050487292c9a094/src/vs/workbench/common/notifications.ts#L328

But for `command:` links, I know when we talked about getting them in hovers there was security concerns and `MarkdownString` was added with the `isTrusted` property. Does the same thing need to be done here? Is that the right approach?
